### DISCUSSION TITLE
Next CP2K Developers Meeting

### DISCUSSION THREAD

**Discussion by robertschade** [cp2k/cp2k#2615] [Team Posts]
Dear https://github.com/orgs/cp2k/teams/cp2k-developers,
we would like to invite all of you to our next developers meeting on February 6th 14-16:00 CET. Please also forward this invitation to other people that might be interested.
If you would like to contribute a topic or question to discuss, please simply add it to the slides at
https://docs.google.com/presentation/d/1eyPfO2Dka9gv_7afuHiPEO5x3neVAg3T1WKzw4Bqzcs/edit?usp=sharing

Video conference information:
https://uni-paderborn-de.zoom.us/j/95704402924?pwd=V3ZTTjhGcDVQQWtDVFRjUTFIQTMrQT09

Meeting ID: 957 0440 2924
Passcode: 310036

Best Wishes

**Comment by RMeli** [comment]
Hello. I just had a look at the slides of the last developers' meeting and noticed that there is some effort to revamp the documentation. Have you ever considered applying to [Google Season of Docs](https://developers.google.com/season-of-docs)? The deadline this year is March 24, 2023 at 18:00 UTC.

**Reply by oschuett** [reply]
Yes, generally that's a good idea. However, I think we should hold off until we're done with the migration to Sphinx.

As I understand it, the Google program is about lifting existing documentation, that's already decent, to a professional level.

**Reply by RMeli** [reply]
> the Google program is about lifting existing documentation, that's already decent, to a professional level

I thought that porting the documentation to Sphinx could be part of GSoD, but I'm not totally sure TBH. I know a project that participated in 2019 and IIRC the technical writer there did set up (and write) the whole Sphinx-based user guide (FYI, see the [MDAnalysis User Guide](https://userguide.mdanalysis.org/stable/index.html#)).

**Comment by oschuett** [comment]
> FYI, see the [MDAnalysis User Guide](https://userguide.mdanalysis.org/stable/index.html#)

This does indeed look very nice!

> I thought that porting the documentation to Sphinx could be part of GSoD, but I'm not totally sure TBH. 

I expect that there will be three parts to our documentation revamp:

1. Technical groundwork:
     - Re-configuring our web-server.
     - Running [Pandoc](https://pandoc.org/) to convert the DokuWiki pages to Markdown.
     - Re-implementing the [reference manual](https://github.com/cp2k/cp2k/blob/master/tools/manual/cp2k_input.xsl) in Python.
     - Setting up search via Algolia.
 
2. Reviewing and curating the existing content.
3. Writing missing content, mostly text-book-style explanatory articles.

I'll be focusing on the technical part 1. So, perhaps you - or someone else - would be interested in driving one of the other parts?

I believe for part 2. and 3. we could indeed really benefit from a professional writer. However, the Google program does require some [time commitment](https://developers.google.com/season-of-docs/docs/timeline) from our side as well.

**Comment by hfp** [comment]
Since we move to Markdown, we have a lot of choices. I personally like https://getgrav.org/ and like Dokuwiki it does not use an obscure database, i.e., MD-files can be copied into folders and are taken from there like any "flat" CMS does. Usually, the folder structure can be leveraged as document structure as well. Of course, there are various themes available. I believe the effort is low getting "a first picture" given we have MD-files.

**Reply by oschuett** [reply]
Oh, GRAV does look like a worthy successor of DokuWiki! Thanks for sharing :-)

However, for the user documentation I was planning to move it into our git repository. My hope is that it'll lower the barrier to contribution and improve quality because of the pull request reviews.
